RobertBeaudenon/SOEN390-CampusGuideMap
# Issues #2, #7, #12, #16, #23 Combined Unit and UI Test Coverage using JaCoCo

#2, #7, #12, #16, #23 :
 
We can now generate a combined Unit and UI test coverage using JaCoCo.

How to view the UI and Unit test in one JaCoCo report?

Step 1) Open Android Studio

Step 2) Click on Terminal (it's located at the bottom left, last option)

Step 3) Enter the command to execute the report (This takes about 1m 30s, the more test we have, the longer the time gets)

Command to execute combined UI and Unit test in one  JaCoCo report:

``gradlew clean combinedTestReportDebug``

Important information: This command relies on the fact that all UI and Unit tests are passing. If one fails, it stops and will give an error on the command line as to which test failed its execution!

What does this command do? 

1) It first runs the UI test (you will see your phone magically opening the app and doing things)

2) It then runs the Unit Test

3) It finally combines the 2 previous test in one JaCoCo report.

The report is stored under (Project View): 
SOEN390-CampusGuideMap\app\build\reports\jacocoCombined\debug\index.html
